pandas/plotting/_matplotlib/core.py

```diff
@@ -1332,7 +1331,11 @@ def __init__(self, data, **kwargs):
                 self.tickoffset = self.bar_width * pos
                 self.lim_offset = 0
 
-        self.ax_pos = self.tick_pos - self.tickoffset
+        if isinstance(self.data.index, ABCMultiIndex):
+            # No real handling for MultiIndex yet
+            self.ax_index = np.arange(len(data))
```

Comment:
Is this the bug you are referring to? Would rather raise than do this if so

Reply:
Raising here would mean no more bar plotting of MultiIndex, which would probably some code for users.
Before the proposed fix, the MultiIndex did renders, but, the level of a multi index are joined as tuple, and the matplotlib method for axis conversion does not handle this, so this fix can only handle single Index.
Breaking an inperfect way of plotting MultiIndex seems a bit harsh.

Reply:
Gotcha. Unfortunately just changing this to an arbitrary range I don't think is a great idea either. Is there no way to maintain existing behavior?

Reply:
This do maintain existing behavior for MultiIndex.
And it improves it for single index

Reply:
i would add a UserWarning here that MultiImdexes are not supported

Reply:
Sure thing. I had to add a `:okwarning:` in doc/source/user_guide/visualization.rst for the doc ci to pass.